File: Astromaximum/src/com/astromaximum/android/view/VcHolder.java
package com.astromaximum.android.view;

import com.astromaximum.android.R;
import com.astromaximum.android.util.Event;

public class VcHolder extends ViewHolder {

	public VcHolder(SummaryItem si) {
		super(si, R.layout.item_vc, LAYOUT_FLAG_TEXT0);
	}

	@Override
	public void fillLayout() {
		Event e = getActiveEvent();
		if (e != null) {
			mText0.setText(e.normalizedRangeString());
			setColorByEventMode(mText0, e);
		} else {
			mText0.setText("");
		}
	}
}

File: Qt/view/VcHolder.cpp
#include "VcHolder.h"
#include "ui_VcHolder.h"
#include "../util/Event.h"

VcHolder::VcHolder()
: ui(new Ui::VcHolder)
{
	ui->setupUi(this);
}

VcHolder::~VcHolder()
{
	delete ui;
}

void VcHolder::fillLayout()
{
	Event* e = activeEvent();
	if (e) {
		ui->mText0->setText(e->normalizedRangeString());
		setColorByEventMode(ui->mText0, e);
	}
	else {
		ui->mText0->setText("");
	}
}
